context/src/main/java/io/opentelemetry/context/LazyStorage.java

```diff
@@ -56,12 +56,16 @@ static ContextStorage get() {
     return storage;
   }
 
+  static void set(ContextStorage storage) {
+    LazyStorage.storage = storage;
+  }
+
   private static final String CONTEXT_STORAGE_PROVIDER_PROPERTY =
       "io.opentelemetry.context.contextStorageProvider";
 
   private static final Logger logger = Logger.getLogger(LazyStorage.class.getName());
 
-  private static final ContextStorage storage;
+  private static ContextStorage storage;
```

Comment:
Intentionally not volatile for now - maybe it's too optimistic to hope `set` is only ever called by a user at the very beginning of the map but unlike `OpenTelemetry` itself this is a very advanced API.

Reply:
An idea that comes to mind is having ResettableContextStorageProvider in the testing package. I would need to add a slightly hacky check for that provider in `LazyStorage.createStorage()` to use it + another normally SPI-loaded or default contextstorage when it's present but it doesn't seem so bad actually. Let me play with it.

Reply:
Was able to remove `set(ContextStorage)` in the main artifact - there is some crosstalk where the prod artifact is aware of the classname of the test helper, and the test helper duplicates some logic from the prod logic, but it's all contained in implementation and seems safe to me. Added `defaultStorage` API which was needed anyways to support custom storage implementations delegating to the standard.